src/ZiraLink.Api.Application/Services/CustomerService.cs

```diff
@@ -157,61 +160,35 @@ public async Task UpdateProfileAsync(string userId, string name, string family,
-            var baseUri = _idsUri;
-            var uri = new Uri(baseUri, "User");
-            var response = await client.PatchAsync(uri.ToString(), content);
+            // var content = new StringContent(JsonSerializer.Serialize(jsonObject), Encoding.UTF8, "application/json");
+            // var baseUri = _idsUri;
+            // var uri = new Uri(baseUri, "User");
+            // var response = await client.PatchAsync(uri.ToString(), content);
+
+            // var responseString = await response.Content.ReadAsStringAsync(cancellationToken);
+            // var userCreationResult = JsonSerializer.Deserialize<ApiResponse<string>>(responseString, new JsonSerializerOptions { PropertyNamingPolicy = JsonNamingPolicy.CamelCase });
 
-            var responseString = await response.Content.ReadAsStringAsync(cancellationToken);
-            var userCreationResult = JsonSerializer.Deserialize<ApiResponse<string>>(responseString, new JsonSerializerOptions { PropertyNamingPolicy = JsonNamingPolicy.CamelCase });
+            var userCreationResult = await _httpTools.CallIDSApis<ApiResponse<string>>(httpClient, "User", jsonObject, HttpMethod.Post, cancellationToken);
             if (!userCreationResult.Status)
-                throw new ApplicationException("Updating profile on identity server failed");
+                throw new ApplicationException("User creation on identity server failed");
 
             customer.Name = name;
             customer.Family = family;
 
             await _dbContext.SaveChangesAsync(cancellationToken);
         }
 
-        private async Task<HttpClient> InitializeHttpClientAsync(CancellationToken cancellationToken)
-        {
-            var client = new HttpClient();
-            var disco = await client.GetDiscoveryDocumentAsync(new DiscoveryDocumentRequest
-            {
-                Address = _idsUri.ToString(),
-                Policy =
-                {
-                    RequireHttps = string.IsNullOrWhiteSpace(_configuration["ZIRALINK_USE_HTTP"]) || bool.Parse(_configuration["ZIRALINK_USE_HTTP"]!) == false
-                }
-            }, cancellationToken);
-            if (disco.IsError)
-                throw new ApplicationException("Failed to get discivery document");
-
-            var tokenResponse = await client.RequestClientCredentialsTokenAsync(new ClientCredentialsTokenRequest
-            {
-                Address = disco.TokenEndpoint,
-
-                ClientId = "back",
-                ClientSecret = "secret",
-                Scope = "ziralink IdentityServerApi"
-            }, cancellationToken);
-
-            if (tokenResponse.IsError)
-                throw new ApplicationException("Failed to get token from identity server");
-
-            client.SetBearerToken(tokenResponse.AccessToken);
-
-            return client;
-        }
+        protected int Count { get; set; }
```

Comment:
Where is Count used in?

Reply:
Removed. This was just to test something. I forgot to remove.